test/py/CMakeLists.txt

```diff
@@ -23,37 +23,71 @@
 #####################################################################################
 
 include(PythonModules)
+set(VENV ${CMAKE_BINARY_DIR}/test/py/venv)
 
-function(add_py_test NAME SCRIPT)
+if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+    file(READ "/etc/os-release" OS_RELEASE_CONTENTS)
+    if(OS_RELEASE_CONTENTS MATCHES "ID=\"sles\"")
+        message("This is SUSE Linux Enterprise Server.")
+        set(REQUIREMENTS ${CMAKE_CURRENT_SOURCE_DIR}/requirements-sles.txt)
+    else()
+        message("This is NOT SUSE Linux Enterprise Server.")
+        set(REQUIREMENTS ${CMAKE_CURRENT_SOURCE_DIR}/requirements.txt)
+    endif()
+endif()
+
+function(add_py_venv_fixture VENV_DIR REQUIREMENTS_FILE)
     foreach(PYTHON_VERSION ${PYTHON_VERSIONS})
-        set (ENV_COMMAND ${CMAKE_COMMAND} -E env 
+        set (ENV_COMMAND ${CMAKE_COMMAND} -E env
             "PYTHONPATH=$<TARGET_FILE_DIR:migraphx_pybind_${PYTHON_VERSION}>"
             "PYTHONMALLOC=debug"
             "MALLOC_CHECK_=3"
         )
         set(PYTHON_EXECUTABLE ${PYTHON_${PYTHON_VERSION}_EXECUTABLE})
+        set(VIRTUAL_ENV_DIR ${VENV_DIR}/${PYTHON_VERSION})
+
+        add_test(NAME py_${PYTHON_VERSION}_initialize_env COMMAND ${PYTHON_EXECUTABLE} -m venv ${VIRTUAL_ENV_DIR} --clear)
+        set_tests_properties(py_${PYTHON_VERSION}_initialize_env PROPERTIES FIXTURES_SETUP ${PYTHON_VERSION}_INIT_VENV)
+        set(ENV{VIRTUAL_ENV} ${VIRTUAL_ENV_DIR})
```

Comment:
Since we are using the path to python in the virtual environment, this might not be needed. Should I remove this?

Reply:
This env variable wont be applied when running the test, so this can be removed.